src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+package mate.academy.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.order.CreateOrderRequestDto;
+import mate.academy.dto.order.OrderDto;
+import mate.academy.dto.order.OrderItemDto;
+import mate.academy.dto.order.UpdateOrderRequestDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.OrderItemMapper;
+import mate.academy.mapper.OrderMapper;
+import mate.academy.model.Order;
+import mate.academy.model.OrderItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.cart.ShoppingCartRepository;
+import mate.academy.repository.order.OrderItemRepository;
+import mate.academy.repository.order.OrderRepository;
+import mate.academy.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(User user, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Can't create order. CartItems is empty");
```

Comment:
It's better to create 
```suggestion
            throw new OrderProcessingException("Can't create order. CartItems is empty");
```